axelor-production/src/main/java/com/axelor/apps/production/service/operationorder/planning/OperationOrderPlanningCommonService.java

```diff
@@ -101,4 +102,26 @@ public void checkIfPlannedStartDateTimeIsBeforeCurrentDateTime(OperationOrder op
           DateTimeFormatter.ISO_DATE_TIME.format(plannedStartDateT));
     }
   }
+
+  public void planWithStrategyAsapOutSourced(OperationOrder operationOrder) {
+    LocalDateTime plannedStartDate = operationOrder.getPlannedStartDateT();
+    LocalDateTime lastOperationDate = operationOrderService.getLastOperationDate(operationOrder);
+    LocalDateTime maxDate = LocalDateTimeHelper.max(plannedStartDate, lastOperationDate);
+
+    operationOrder.setPlannedStartDateT(maxDate);
+    operationOrder.setPlannedDuration(operationOrderService.getOutsourcingDuration(operationOrder));
+    operationOrder.setPlannedEndDateT(
+        operationOrder.getPlannedStartDateT().plusSeconds(operationOrder.getPlannedDuration()));
+  }
+
+  public void planWithStrategyAtTheLatestOutSourced(OperationOrder operationOrder) {
+    LocalDateTime plannedEndDate = operationOrder.getPlannedEndDateT();
+    LocalDateTime nextOperationDate = operationOrderService.getNextOperationDate(operationOrder);
+    LocalDateTime minDate = LocalDateTimeHelper.min(plannedEndDate, nextOperationDate);
+
+    operationOrder.setPlannedEndDateT(minDate);
+    operationOrder.setPlannedDuration(operationOrderService.getOutsourcingDuration(operationOrder));
+    operationOrder.setPlannedStartDateT(
+        operationOrder.getPlannedEndDateT().minusSeconds(operationOrder.getPlannedDuration()));
+  }
```

Comment:
these methods are called only from child classes, so please change their visibility to `protected`